duckduckgo/Android
# Feature/detecting keyboard dismissal

Asana Issue URL: https://app.asana.com/0/471012189846020/487962201286661

## Description
Added custom `EditText` subclass which is able to detect when the keyboard is being dismissed using the back button. Using this, we can react to when the user hits to dismiss the keyboard but hasn't searched for anything yet, and navigate them back to the landing page.

This PR also tidies up some of the UX around focussing and keyboard showing. You should no longer have to tap twice on the URL to get the keyboard to appear.


## Steps to Test this PR:
1. From landing page, tap the text box. Dismiss the keyboard, ensuring you return to the landing page
1. From landing page, tap the text box. Tap outside of the omnibar. Ensure you are returned to the landing page
1. From landing page, tap the text box. type a search and hit enter. Repeat the 2 steps above and ensure you are NOT returned to the landing page.
